### PR TITLE
Hardwaremonitor: Improve PI hardware detection

### PR DESCRIPTION
**Pull request by akamming**
Pi hardware detection is now based on the succesful execution of vcgencmd command instead of checking for specific models.

### CODE REVIEW

**Comment by kofec** [comment]
Hi,
What if vcgencmd is in different localization ? is It possible to do it more generic ?
```
which vcgencmd 
/usr/bin/vcgencmd
```

**Comment by akamming** [comment]
> Hi,
> What if vcgencmd is in different localization ? is It possible to do it more generic ?
> which vcgencmd 
> /usr/bin/vcgencmd

Wat would cause vcgencmd to be in a different location? I think in Raspbian it is always in the same place….

**Comment by kofec** [comment]
I think there is more than one Linux distribution :-)
In my case it is openwrt

**Comment by akamming** [comment]
made it more generic.  now detects is vcgencmd is in the search path.

Still works on pi 4 and pi 3b+

@kofec can/will you test on openwrt?

**Comment by gizmocuz** [comment]
@akamming , maybe for safety reasons don't use std::atof if you expect the value to be a non decimal but use
	float voltage = static_cast<float>(atof(tmpline.c_str()));

This also eliminates the try/catch warning
else for the catch use (...) to solve a compile warning

HardwareMonitor.cpp(284,34): warning C4101: 'e': unreferenced local variable
HardwareMonitor.cpp(321,34): warning C4101: 'e': unreferenced local variable
HardwareMonitor.cpp(358,34): warning C4101: 'e': unreferenced local variable

**Comment by akamming** [comment]
@gizmocuz  I don't have this warning when i compile.  is this on windows?

Anyway: I change the code in latest commit to avoid use of std::atof..

**Comment by gizmocuz** [comment]
@akamming , thanks! Yes this is on Windows.... It's great to compile on multiple platforms to eliminate compile issues ;)

File: main/domoticz.cpp [hunk 1]
```diff
@@ -124,12 +124,12 @@ std::string szWebRoot;
 std::string dbasefile;
 
 bool bHasInternalTemperature=false;
-std::string szInternalTemperatureCommand = "/opt/vc/bin/vcgencmd measure_temp";
+std::string szInternalTemperatureCommand = "vcgencmd measure_temp";
```
Comment (kiddigital) [inline main/domoticz.cpp]: @akamming , maybe separate the command from the arguments? Have a define at the top for the command and use this define here.

File: main/domoticz.cpp [hunk 2]
```diff
@@ -124,12 +124,12 @@ std::string szWebRoot;
...
 
 bool bHasInternalClockSpeeds=false;
-std::string szInternalARMSpeedCommand = "";
-std::string szInternalV3DSpeedCommand = "";
-std::string szInternalCoreSpeedCommand = "";
+std::string szInternalARMSpeedCommand = "vcgencmd measure_clock arm";
+std::string szInternalV3DSpeedCommand = "vcgencmd measure_clock v3d";
+std::string szInternalCoreSpeedCommand = "vcgencmd measure_clock core";
```
Comment (kiddigital) [inline main/domoticz.cpp]: same here.. use define for command

File: main/domoticz.cpp [hunk 3]
```diff
@@ -468,46 +468,31 @@ void GetAppVersion()
...
-				szInternalV3DSpeedCommand = "/opt/vc/bin/vcgencmd measure_clock v3d";
-				szInternalCoreSpeedCommand = "/opt/vc/bin/vcgencmd measure_clock core";
-				bHasInternalClockSpeeds=true;
-
-				break;
-			}
+	//Check if we have vcgencmd (are running on a RaspberryPi)
+	//
+	int returncode=0;
+	std::vector<std::string> ret = ExecuteCommandAndReturn ("vcgencmd measure_temp",returncode);
```
Comment (kiddigital) [inline main/domoticz.cpp]: @akamming , here you have the command hard-coded and are not using the defined variables. Could go wrong now if this works but variables are different.

File: main/domoticz.cpp [hunk 4]
```diff
@@ -468,46 +468,31 @@ void GetAppVersion()
...
+			_log.Log(LOG_STATUS,"Hardware Monitor: Raspberry Pi detected");
+			//Core temperature of BCM2835 SoC
+			szInternalTemperatureCommand = "vcgencmd measure_temp";
+			bHasInternalTemperature = true;
 
+			//PI Clock speeds	
+			szInternalARMSpeedCommand = "vcgencmd measure_clock arm";
+			szInternalV3DSpeedCommand = "vcgencmd measure_clock v3d";
+			szInternalCoreSpeedCommand = "vcgencmd measure_clock core";
+			bHasInternalClockSpeeds=true;
```
Comment (kiddigital) [inline main/domoticz.cpp:497]: Above are hard-coded again... use the defined variables I would think

**Review by kiddigital — COMMENTED** [review]
@akamming , some suggestions :)

**Comment by akamming** [comment]
@kiddigital sure no problem....just added the commit

**Comment by kofec** [comment]
Hi, It is working on openwrt:
image

**Comment by akamming** [comment]
@kofec  Tx! 
@gizmocuz all work is done then. i think it can be merged...

**Comment by gizmocuz** [comment]
Thanks!!

**Pull request merged**